src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -76,12 +87,25 @@ public void visitToken(DetailAST ast) {
             final DetailAST switchAST = defaultGroupAST.getParent();
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
-
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
-                || defaultGroupAST.getColumnNo()
-                    != lastGroupAST.getColumnNo()) {
+            if (!skipIfLastAndSharedWithCaseInSwitch(ast)
+                    && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
```

Comment:
@romani Why are we checking line number and column numbers here in original logic of check?
Shouldn't we just be checking if `defaultGroupAST` != `lastGroupAST`?